src/common/Utilities.py

```diff
@@ -464,28 +464,41 @@ def str_to_int_list(str_list):
         int_list = int_list + int_item
     return int_list
 
-def dot_product_test(operator, num_tests = 20, max_err = 10e-5):
+def test_operator_adjoint(operator, num_tests = 20, max_err = 10e-5):
     '''
     Test if a given operator is adjoint.
-    The operator needs to have been already set_up() with valid AcquisitionModel and ImageData objects.
+    The operator needs to have been already set_up() with valid objects.
     The operator needs to have methods direct() and adjoint() implemented
+
+    Parameters
+    ----------
+    operator  :
+        Any SIRF operator that implements direct() and adjoint()
+    num_tests : int, optional
+        Number of tests with random data that will be executed. Default 20
+    max_err   : double, optional
+        Maximum allowed normalized error, tolerance. Change not recommended. Default 10e-5
     '''
-    # Get image and data to know the sizes of each operation.
-    image=operator.domain_geometry()
-    image.fill(1.0)
-    data=operator.direct(image) # Is there a better way than this? Similar to .domain_geomerty() ?
+    # Get x to know the size of each operation (we can not currently get y, so it will be obtained in the loop)
+    x=operator.domain_geometry()
     for __ in range(num_tests):
-        # generate random data
-        x0 = numpy.random.rand(*image.shape) * 10**numpy.random.randint(-2, 3)
-        y0 = numpy.random.rand(*data.shape ) * 10**numpy.random.randint(-2, 3)
-        # fill image and data
-        image.fill(x0)
-        data.fill(y0)
-        # compute direct and adjoint
-        y_hat = operator.direct(image)
-        x_hat = operator.adjoint(data)
+        ## generate random data and direct()
+        x0 = numpy.random.rand(*x.shape) * 10**numpy.random.randint(-2, 3)
+        # add a complex part if necessary
+        if numpy.any(numpy.iscomplex(x.as_array())):
```

Comment:
this can presumably never be true, as all its elements are real by construction. I believe you need `numpy.iscomplexobj`